### PR TITLE
[0.10.0] Elective List With API

### PR DESCRIPTION
**Pull request by SiravitPhokeed**
**Features**
- Learn Electives page data is now from the API
  Resolves FRO-74
- Grayed out base path in server-side fetch log

**Fixes**
- Schedule Glance erroneously shows the first period from midnight to assembly
- Server-side fetch log not indented

### CODE REVIEW

File: utils/backend/mysk/createMySKClient.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export default async function createMySKClient(
     fetch: async <Data extends {} | unknown = unknown>(
       path: Parameters<typeof fetchMySKAPI>["0"],
       options?: Parameters<typeof fetchMySKAPI>["2"],
-    ): Promise<FetchReturn<Data> | null> =>
```
Comment (pixelpxed) [inline utils/backend/mysk/createMySKClient.ts:44]: Why is this line removed?

File: components/elective/ElectiveListItem.tsx [hunk 2]
```diff
@@ -16,18 +16,24 @@ import {
...
+ * @param enrolled Whether the Student is enrolled in the Elective Subject.
  * @param onRadioToggle Triggers when the radio button is toggled.
  * @param onClick Triggers when the item is clicked.
  */
 const ElectiveListItem: StylableFC<{
   electiveSubject: ElectiveSubject;
-  selected?: boolean;
+  radioSelected?: boolean;
+  detailSelected?: boolean;
+  enrolled?: boolean;
```
Comment (pixelpxed) [inline components/elective/ElectiveListItem.tsx:29]: Having these list items as a state in the element would surely make it handy to get all the info in the future if needed to develop more features and see UI bugs if present.

**Review by pixelpxed — APPROVED** [review]
LGTM, one small minor question though.

File: utils/backend/mysk/createMySKClient.ts (hunk 1 above)
Comment (Galax028) [reply utils/backend/mysk/createMySKClient.ts:44]: Because the API will always return data in JSON even if there is an error, so `null` can be safely removed

**Review by smartwhatt — APPROVED** [review]
LGTM!
